### PR TITLE
Fix auto turn ai bug

### PR DESCRIPTION
**Pull request by JordanRoth**
This PR fixes the bug where the game would crash when the AI tried to determine villains after the lanes auto moved due to the timer expiring.

Note: Let me know if you experience slight slowness. Ply depth 5 works for me on Release but is a little slow on debug, so I put it down to ply depth 3.

### CODE REVIEW

File: src/ai.hpp [hunk 1]
```diff
@@ -21,11 +21,11 @@ struct PlayerOutcome
 class AI
 {
 public:
-	const int MaxPly = 5;
+	static const int MaxPly = 3;
```
Comment (JordanRoth) [inline src/ai.hpp:24]: This was the source of the bug, somehow. Apparently, if you use a non-static constant, C++ just randomly assigns a value of it, so the MaxPly was essentially undefined and thus one of the AI base cases was never fulfilled.

File: src/lane_manager.cpp [hunk 2]
```diff
@@ -33,10 +33,14 @@ bool LaneManager::update(float ms)
...
-	lane_queue(m_lanes[direction::WEST], m_lane_coords[direction::EAST], ms);
-	lane_queue(m_lanes[direction::SOUTH], m_lane_coords[direction::SOUTH], ms);
-	lane_queue(m_lanes[direction::EAST], m_lane_coords[direction::WEST], ms);
+	if (lane_queue(m_lanes[direction::NORTH], m_lane_coords[direction::NORTH], ms) ||
+		lane_queue(m_lanes[direction::WEST], m_lane_coords[direction::EAST], ms) ||
+		lane_queue(m_lanes[direction::SOUTH], m_lane_coords[direction::SOUTH], ms) ||
+		lane_queue(m_lanes[direction::EAST], m_lane_coords[direction::WEST], ms)) {
+		// lane_queue returns true if its time has expired.
+		// If this is the case, we should readjust new villains.
+		m_ai->make_villains_decide(m_lanes);
```
Comment (JordanRoth) [inline src/lane_manager.cpp:42]: Efficiency hack

File: src/ai.hpp (hunk 1 above)
Comment (PenguinFlyingHigh) [reply src/ai.hpp:24]: How'd you figure this one out?

**Review by theochit — APPROVED** [review]

File: src/ai.hpp (hunk 1 above)
Comment (JordanRoth) [reply src/ai.hpp:24]: I noticed that the value of MaxPly was a hugely negative number whose exact value changed each time I ran the program. It took ages to track it to that point but I realized then that it wasn't retrieving the constant value.

Most confusingly is that VillainSpawnProbability constant seems to work despite its similar declaration.

**Pull request merged**